### PR TITLE
Coupon finder for dominos

### PR DESCRIPTION
**Pull request by R0bk**
Added a coupon finder for dominos, using the same coupon search engine that ozbargin uses (https://www.couponese.com/store/dominos.com.au/).

### CODE REVIEW

File: scripts/dominos.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+module.exports = function (robot) {
+
+    function printItems(res, item) {
+        var noOfCoupons = 10;
+        var specialWords = false;
+        var response = "";
+	var expiry = false;
+	
+        if (item.match(/-n/g) !== null) {
```
Comment (mitchmcdee) [inline scripts/dominos.js]: What do these regex do exactly? maybe write some comments?

File: scripts/dominos.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+var cheerio = require("cheerio");
+
+module.exports = function (robot) {
+
+    function printItems(res, item) {
+        var noOfCoupons = 10;
+        var specialWords = false;
+        var response = "";
+	var expiry = false;
```
Comment (mitchmcdee) [inline scripts/dominos.js]: indenting

File: scripts/dominos.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+        if (item.match(/-n/g) !== null) {
+            noOfCoupons = item.match(/-n\s+\d{0,3}/g)[0].match(/\d+/)[0];
+        }
+
+        if (item.match(/-f/g) !== null) {
+            specialWords = item.match(/-f\s+[^\-]+/g)[0].match(/[^\-f]+/g)[0].trim();
+            specialWords = new RegExp(specialWords, "gi");
+        }
+
+	if (item.match(/-e/g) !== null) {
```
Comment (mitchmcdee) [inline scripts/dominos.js]: indenting and explanation maybe

File: scripts/dominos.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+                if (noOfCoupons > 0) {
+                    var code = $(this).find(".ov-code").find("strong").text();
+                    var message = $(this).find(".ov-title").find("a").text();
+		    var expDate = $(this).find(".ov-expiry").text().trim();
+		    
+		    if (expiry) {
+			message = expDate + "\t" + message;
+	 	    }		
+
+                    if (!!specialWords) {
```
Comment (mitchmcdee) [inline scripts/dominos.js]: if(!!specialWord && message.match(specialWords) !== null) would suffice here if you extracted out noOfCoupons-- (which is performed in every statement)

File: scripts/dominos.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+                    }
+                }
+            });
+            res.send(response);
+        });
+    }
+
+    // Response if no args given
+    robot.respond(/!?dominos(.*)$/, function (res) {
+	res.send("Fetching the latest dominos coupons");
```
Comment (mitchmcdee) [inline scripts/dominos.js]: indenting

File: scripts/dominos.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+
+        var URL = "https://www.couponese.com/store/dominos.com.au/";
+        robot.http(URL).get()(function (err, resp, body) {
+            $ = cheerio.load(body);
+            $(".ov-coupon:not(.ov-expired)").each(function (i, elem) {
+                if (noOfCoupons > 0) {
+                    var code = $(this).find(".ov-code").find("strong").text();
+                    var message = $(this).find(".ov-title").find("a").text();
+		    var expDate = $(this).find(".ov-expiry").text().trim();
```
Comment (mitchmcdee) [inline scripts/dominos.js]: indenting all dis section

File: scripts/dominos.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+//        if -n isn't specified the number of coupons listed is 10 or as many as can be found up to 10
+//
+
+
+var cheerio = require("cheerio");
+
+module.exports = function (robot) {
+
+    function printItems(res, item) {
+        var noOfCoupons = 10;
```
Comment (mitchmcdee) [inline scripts/dominos.js]: provide a small comment on these variables maybe?

**Comment by mitchmcdee** [comment]
Also looks like a lot of tab/space indenting mash ups-- doing a bulk fix of that would fix most of the indenting I'm guessing

**Comment by TRManderson** [comment]
Multiline help is broken. I know it's bad, but can you put the stuff from commands on one line?

**Pull request merged**